homeassistant/components/volkswagencarnet/__init__.py

```diff
@@ -0,0 +1,237 @@
+
+from datetime import timedelta
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD, CONF_RESOURCES, CONF_NAME, CONF_SCAN_INTERVAL)
+from homeassistant.helpers import discovery
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.icon import icon_for_battery_level
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'volkswagencarnet'
+DATA_KEY = DOMAIN
+CONF_REGION = 'region'
+DEFAULT_REGION = 'SV'
+CONF_MUTABLE = 'mutable'
+
+REQUIREMENTS = ['volkswagencarnet==4.0.20']
+
+SIGNAL_STATE_UPDATED = '{}.updated'.format(DOMAIN)
+
+MIN_UPDATE_INTERVAL = timedelta(minutes=1)
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=5)
+
+COMPONENTS = {
+    'sensor': 'sensor',
+    'binary_sensor': 'binary_sensor',
+    'lock': 'lock',
+    'device_tracker': 'device_tracker',
+    'switch': 'switch',
+}
+
+RESOURCES = [
+    'position',
+    'distance',
+    'climatisation',
+    'window_heater',
+    'charging',
+    'battery_level',
+    'fuel_level',
+    'service_inspection',
+    'oil_inspection',
+    'last_connected',
+    'charging_time_left',
+    'electric_range',
+    'combustion_range',
+    'combined_range',
+    'charge_max_ampere',
+    'climatisation_target_temperature',
+    'external_power',
+    'parking_light',
+    'climatisation_without_external_power',
+    'door_locked',
+    'trunk_locked',
+    'request_in_progress'
+]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_REGION, default=DEFAULT_REGION): cv.string,
+        vol.Optional(CONF_MUTABLE, default=True): cv.boolean,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_UPDATE_INTERVAL): (
+            vol.All(cv.time_period, vol.Clamp(min=MIN_UPDATE_INTERVAL))),
+        vol.Optional(CONF_NAME, default={}): vol.Schema(
+            {cv.slug: cv.string}),
+        vol.Optional(CONF_RESOURCES): vol.All(
+            cv.ensure_list, [vol.In(RESOURCES)])
+    }),
+}, extra = vol.ALLOW_EXTRA)
+
+def setup(hass, config):
+    """Setup Volkswagen Carnet component"""
+    interval = config[DOMAIN].get(CONF_SCAN_INTERVAL)
+    data = hass.data[DATA_KEY] = VolkswagenData(config)
+    
+    from volkswagencarnet import Connection
+
+    _LOGGER.debug("Creating connection to volkswagen carnet")
+    connection = Connection(
+        username = config[DOMAIN].get(CONF_USERNAME),
+        password = config[DOMAIN].get(CONF_PASSWORD),
+        #region = config[DOMAIN].get(CONF_REGION) # add support for this
+    )
+
+    # login to carnet
+    _LOGGER.debug("Logging in to volkswagen carnet")
+    connection._login()
+    if not connection.logged_in:
+        _LOGGER.warning('Could not login to volkswagen carnet, please check your credentials')
+
```

Comment:
We should `return False` if we can't log in due to bad credentials, since it's a non recoverable error.